src/query/expression/src/values.rs

```diff
@@ -286,6 +288,85 @@ impl<'a> ScalarRef<'a> {
     }
 }
 
+impl PartialOrd for Scalar {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.as_ref().partial_cmp(&other.as_ref())
+    }
+}
+
+impl PartialOrd for ScalarRef<'_> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        match (self, other) {
+            (ScalarRef::Null, ScalarRef::Null) => Some(Ordering::Equal),
+            (ScalarRef::EmptyArray, ScalarRef::EmptyArray) => Some(Ordering::Equal),
+            (ScalarRef::Number(n1), ScalarRef::Number(n2)) => n1.partial_cmp(n2),
+            (ScalarRef::Boolean(b1), ScalarRef::Boolean(b2)) => b1.partial_cmp(b2),
+            (ScalarRef::String(s1), ScalarRef::String(s2)) => s1.partial_cmp(s2),
+            (ScalarRef::Timestamp(t1), ScalarRef::Timestamp(t2)) => t1.partial_cmp(t2),
+            (ScalarRef::Array(a1), ScalarRef::Array(a2)) => a1.partial_cmp(a2),
+            (ScalarRef::Tuple(t1), ScalarRef::Tuple(t2)) => t1.partial_cmp(t2),
+            (ScalarRef::Variant(v1), ScalarRef::Variant(v2)) => v1.partial_cmp(v2),
```

Comment:
Need cast into Json and CMP it with JSON, you can add todo here.

Reply:
Is JSON implemented now?

Reply:
no, so a todo comment needed.

Reply:
OK